### PR TITLE
ci: Minor improvements inspired by Bitcoin Core

### PR DESCRIPTION
**Pull request by real-or-random**
This steals two recent CI improvements from bitcoin/bitcoin.  See individual commit messages.

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -53,11 +55,11 @@ cat_logs_snippet: &CAT_LOGS
 
 merge_base_script_snippet: &MERGE_BASE
   merge_base_script:
-    - if [ "$CIRRUS_PR" = "" ]; then exit 0; fi
-    - git fetch $CIRRUS_REPO_CLONE_URL $CIRRUS_BASE_BRANCH
     - git config --global user.email "ci@ci.ci"
     - git config --global user.name "ci"
```
Comment (hebasto) [inline .cirrus.yml:59]: Those lines could be [removed](https://github.com/bitcoin/bitcoin/commit/ac1d99240af6c5d3ed5db2beea1479903d949a37) as well.

Comment (real-or-random) [reply .cirrus.yml:59]: Oh, I see. Do you want to open a PR?

Comment (hebasto) [reply .cirrus.yml:59]: > Oh, I see. Do you want to open a PR?

I'll handle it :)

Comment (hebasto) [reply .cirrus.yml:59]: > Do you want to open a PR?

#1201.

**Pull request merged**